aci/resource_aci_fvesg_test.go

```diff
@@ -0,0 +1,183 @@
+package aci
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ciscoecosystem/aci-go-client/client"
+	"github.com/ciscoecosystem/aci-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAciEndpointSecurityGroup_Basic(t *testing.T) {
+	var endpoint_security_group models.EndpointSecurityGroup
+	description := "endpoint_security_group created while acceptance testing"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAciEndpointSecurityGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAciEndpointSecurityGroupConfig_basic(description),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAciEndpointSecurityGroupExists("aci_endpoint_security_group.fooendpoint_security_group", &endpoint_security_group),
+					testAccCheckAciEndpointSecurityGroupAttributes(description, &endpoint_security_group),
+				),
+			},
+			{
+				ResourceName:      "aci_endpoint_security_group",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAciEndpointSecurityGroup_update(t *testing.T) {
+	var endpoint_security_group models.EndpointSecurityGroup
+	description := "endpoint_security_group created while acceptance testing"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAciEndpointSecurityGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAciEndpointSecurityGroupConfig_basic(description),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAciEndpointSecurityGroupExists("aci_endpoint_security_group.fooendpoint_security_group", &endpoint_security_group),
+					testAccCheckAciEndpointSecurityGroupAttributes(description, &endpoint_security_group),
+				),
+			},
+			{
+				Config: testAccCheckAciEndpointSecurityGroupConfig_basic(description),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAciEndpointSecurityGroupExists("aci_endpoint_security_group.fooendpoint_security_group", &endpoint_security_group),
+					testAccCheckAciEndpointSecurityGroupAttributes(description, &endpoint_security_group),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAciEndpointSecurityGroupConfig_basic(description string) string {
+	return fmt.Sprintf(`
+
+	resource "aci_endpoint_security_group" "fooendpoint_security_group" {
+		  application_profile_dn  = "${aci_application_profile.example.id}"
+		description = "%s"
+		
+		name  = "example"
+		  annotation  = "example"
+		  exception_tag  = "example"
+		  flood_on_encap  = "example"
+		  match_t  = "example"
+		  name_alias  = "example"
+		  pc_enf_pref  = "example"
+		  pref_gr_memb  = "example"
+		  prio  = "example"
+		  userdom  = "example"
+		}
+	`, description)
+}
```

Comment:
Test file should have the valid values in all the parameters, else it will not pass the tests.